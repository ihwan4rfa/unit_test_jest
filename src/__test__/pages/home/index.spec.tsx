import React from "react";
import { fireEvent, render } from "@testing-library/react";
import HomePage from "@/components/pages/Home";

describe("render home", () => {
  it("should render home", () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
  it("should click button submit on home page", () => {
    const { getByText } = render(<HomePage />);
    const buttonSubmit = getByText("Submit");
    fireEvent.click(buttonSubmit);
    expect(buttonSubmit).toBeInTheDocument();
  });
});
