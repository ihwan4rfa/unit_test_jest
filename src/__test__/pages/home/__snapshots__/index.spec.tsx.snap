// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home should render home 1`] = `
<div>
  <div class="">
    <h1>Add User</h1>
    <br>
    <p>Form</p>
    <br>
    <label for="name">name</label>
    <br>
    <input id="name" type="text">
    <br>
    <br><label for="email">email</label>
    <br><input id="email" type="email">
    <br><br>
    <label for="password">password</label>
    <br>
    <input id="password" type="password">
    <br><br>
    <button class="primary">Save</button>
  </div>
</div>
`;
